tools/c7n_gcp/c7n_gcp/resources/resourcemanager.py

```diff
@@ -48,36 +49,122 @@ def _verb_arguments(self, resource):
-    """GCP resource: https://cloud.google.com/resource-manager/reference/rest/v1/folders
+    """GCP resource: https://cloud.google.com/resource-manager/reference/rest/v2/folders
     """
     class resource_type(TypeInfo):
         service = 'cloudresourcemanager'
         version = 'v2'
         component = 'folders'
         scope = 'global'
-        enum_spec = ('list', 'folders', None)
+        enum_spec = ('search', 'folders', None)
         name = id = 'name'
         default_report_fields = [
             "name", "displayName", "lifecycleState", "createTime", "parent"]
         asset_type = "cloudresourcemanager.googleapis.com/Folder"
+        asset_history = False  # supported but its loses info
         perm_service = 'resourcemanager'
+        permissions = ('resourcemanager.folders.get',)
 
-    def get_resources(self, resource_ids):
-        client = self.get_client()
-        results = []
-        for rid in resource_ids:
-            if not rid.startswith('folders/'):
-                rid = 'folders/%s' % rid
-            results.append(client.execute_query('get', {'name': rid}))
-        return results
+    source_mapping = {'inventory': FolderInventory, 'describe-gcp': DescribeSource}
 
     def get_resource_query(self):
         if 'query' in self.data:
             for child in self.data.get('query'):
                 if 'parent' in child:
-                    return {'parent': child['parent']}
+                    return {'body': {"query": "parent=%s" % child['parent']}}
+                if 'scope' in child:
+                    return {'scope': child['scope']}
+
+
+class ProjectSource:
+
+    def get_resources(self, query):
+        if query and 'subtree' in query:
+            return self.get_sub_tree(query)
+        return super().get_resources(query)
+
+
+class ProjectInventoryHierarchy(ProjectSource, AssetInventory):
+    # for inventory we filter hierarchy client side as this is much faster
+    # for deeply nested trees as there is enough metadata from inventory
+    # to just to do a set intersection for hierarchy.
+
+    def _describe_format(self, resources):
+        for r in resources:
+            # check if operating on cached resources
+            if not self._common_describe_format(r):
+                continue
+            # only while project resource is using v1 of resourcemanager api
+            r['lifecycleState'] = r.pop('state')
+            r['name'] = r.pop('displayName', None)
+            attrs = r.pop('additionalAttributes', None)
+            if attrs:
+                r.update(attrs)
+            # can obsolete as we move to v3 of resourcemanager api
+            if isinstance(r['parent'], str):
+                ptype, pname = r['parent'].split('/', 2)
+                r['parent'] = {'type': ptype[:-1], 'id': pname}
+        return resources
+
+    def _filter_resource_parent(self, resources, parent):
+        return [r for r in resources if parent in r.get('folders', ())]
```

Comment:
maybe should be ```[r for r in resources if parent.split('/')[-1] in r.get('parent', {}).get('id')]```
because when subtree is specified in query the resource has format ```'parent': {'type': 'folder', 'id': '1111111111111'}```

Reply:
this isn't looking at parent, which is a singleton of direct, folders is only available on assetinventory and is full lineage to root with form ['folders/id', 'folders/id-b']